### PR TITLE
[Unified MoE Layer]: Support GLM4.5

### PR DESCRIPTION
**Pull request by hushenwei2000**
### PR types
New features

### PR changes
Models

### Description
- The unified MoE Layer has been adapted for the GLM4.5 model.
- A new YAML configuration item, `use_unified_moe`, has been added. The new MoE Layer will only be used when this setting is enabled. If not configured by the user, it defaults to False, maintaining the use of the old code.

#### 测试
The following configurations for the GLM4.5 model SFT have been verified:
| Model          | Loss Alignment | GSM8K Inference Accuracy |
|---------------|-----------|------------------------------------|
| 1-node EP4TP4PP2SP   | ✅        | Not yet validated                           |
| 8-node EP8TP8PP8SP   | ✅        | Not yet validated                             |

Qwen3MoE retesting has passed.

image

#### TODO
- Integrate follow-up optimization PRs for GLM4.5

### CODE REVIEW

File: paddleformers/transformers/qwen3_moe/modeling.py [hunk 1]
```diff
@@ -341,6 +341,7 @@ def __init__(self, config: Qwen3MoeConfig, layer_idx: int):
                     train_topk_method="greedy",
                     inference_topk_method="greedy",
                     drop_tokens=False,
+                    transpose_gate_weight=False,
```
Comment (lugimzzz) [inline paddleformers/transformers/qwen3_moe/modeling.py:344]: qwen3现在支不支持开aux loss？

Comment (hushenwei2000) [reply paddleformers/transformers/qwen3_moe/modeling.py:344]: Qwen3 组网现在的写法是对齐 Transformers 不返回 router_logits，也就是没有 aux loss 计算。要支持的话得改组网，然后这个新版 MoE Layer 也要加个参数支持。

**Review by lugimzzz — APPROVED** [review]
LGTM
